bundles/org.openhab.binding.flume/src/main/java/org/openhab/binding/flume/internal/handler/FlumeDeviceHandler.java

```diff
@@ -0,0 +1,501 @@
+            query.units = imperialUnits ? FlumeApi.API_UNITS_IMPERIAL : FlumeApi.API_UNITS_METRIC;
+
+            listQuery.add(query);
+        }
+
+        if (System.nanoTime() > this.expiryUsageAlertFetch) {
+            fetchUsageAlerts();
+            this.expiryUsageAlertFetch = System.nanoTime() + USAGE_QUERY_FETCH_INTERVAL.toNanos();
+        }
+
+        if (this.isLinked(CHANNEL_DEVICE_CUMULATIVEUSAGE)
+                && ((System.nanoTime() > this.expiryCumulativeUsage) || forceUpdate)) {
+            FlumeApiQueryWaterUsage query = new FlumeApiQueryWaterUsage();
+
+            query.bucket = FlumeApi.BucketType.DAY;
+            query.untilDateTime = now;
+            query.sinceDateTime = now.withDayOfYear(1);
+            query.groupMultiplier = 400;
+            query.operation = FlumeApi.OperationType.SUM;
+            query.requestId = QUERY_ID_YEAR_TO_DATE;
+            query.units = imperialUnits ? FlumeApi.API_UNITS_IMPERIAL : FlumeApi.API_UNITS_METRIC;
+
+            listQuery.add(query);
+        }
+
+        if (listQuery.isEmpty()) {
+            return;
+        }
+
+        result = getApi().queryUsage(config.id, listQuery);
+
+        if (result == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "@text/offline.cloud-connection-issue");
+            return;
+        }
+
+        Map<String, List<FlumeApiQueryBucket>> queryData = result.get(0);
+        List<FlumeApiQueryBucket> queryBuckets;
+
+        queryBuckets = queryData.get(QUERY_ID_CUMULATIVE_START_OF_YEAR);
+        if (queryBuckets != null) {
+            cumulativeStartOfYear = queryBuckets.get(0).value;
+            startOfYear = now.withDayOfYear(1);
+        }
+
+        queryBuckets = queryData.get(QUERY_ID_YEAR_TO_DATE);
+        if (queryBuckets != null) {
+            cumulativeUsage = queryBuckets.get(0).value + cumulativeStartOfYear;
+            updateState(CHANNEL_DEVICE_CUMULATIVEUSAGE,
+                    new QuantityType<>(cumulativeUsage, imperialUnits ? ImperialUnits.GALLON_LIQUID_US : Units.LITRE));
+            this.expiryCumulativeUsage = System.nanoTime() + this.refreshIntervalCumulative.toNanos();
+        }
+    }
+
+    protected void tryGetCurrentFlowRate(boolean forceUpdate)
+            throws FlumeApiException, IOException, InterruptedException, TimeoutException, ExecutionException {
+        if (this.isLinked(CHANNEL_DEVICE_INSTANTUSAGE)
+                && ((System.nanoTime() > this.expiryInstantUsage) || forceUpdate)) {
+            FlumeApiCurrentFlowRate currentFlowRate = getApi().getCurrentFlowRate(config.id);
+            if (currentFlowRate == null) {
+                return;
+            }
+
+            instantUsage = currentFlowRate.gpm;
+            updateState(CHANNEL_DEVICE_INSTANTUSAGE, new QuantityType<>(instantUsage,
+                    isImperial() ? ImperialUnits.GALLON_PER_MINUTE : Units.LITRE_PER_MINUTE));
```

Comment:
I wonder if this is right. The API returns `GPM` as in `GALLON_PER_MINUTE`? this value is taken and depending on the openHAB configuration a `UNIT` is added. The value is not recalculated when the unit is set to `LITRE_PER_MINUITE`

Reply:
@lsiepel, good catch.  I assumed that the flume api was just sending back the rate based on its region and they had simply labeled the field gpm.  I can't really test it, but I bet you are right that it is specifically gpm.

Is it better to just updateState with a QuantityType of GPM, or should i do the conversion of QuantiyType from GPM to LPM in the binding?

Reply:
It doesn't matter what you feed into the framework with UoM as long as you are sure the value and the unit belong together.  Either hardcoded (always GPM?) or based on the API regional setting.